agent_handlers.go

```diff
@@ -41,39 +41,20 @@ func (a *Agent) onConnectionStateChange(s ConnectionState) {
 	}
 }
 
-func (a *Agent) startOnConnectionStateChangeRoutine() {
-	go func() {
-		for {
-			// CandidatePair and ConnectionState are usually changed at once.
-			// Blocking one by the other one causes deadlock.
```

Comment:
Are these comments ok to be removed?

Reply:
Thanks, no probably not.. The agents behaviour in respect to calling handlers is anyway pretty strange to me.. :(
E.g. do you have an explanation why the connection state handler is called asynchronously in its own Goroutine, while other handlers arnt?

https://github.com/pion/ice/pull/563/files#diff-033dc53d3d924a20f1f9a13378ce32d1461740b98aab338e3cc891bcdca387abR52


I will add the comments again..

Reply:
It seems be added by myself in this PR: https://github.com/pion/ice/pull/225
iirc, there was a deadlock bug before that and I fixed it

Reply:
I've added the comments back here https://github.com/pion/ice/pull/563/files#diff-07afbb40d3dc2699797d28c0689ba544f67ca642d0d3efcff3c02a0bbb3604fdR362